external_plugins/track_and_trace/track_and_trace_plugin_export.cpp

```diff
@@ -7,10 +7,14 @@ namespace sophiatx { namespace plugins { namespace track_and_trace_plugin {
 #pragma clang diagnostic ignored "-Wreturn-type-c-linkage"
 #endif
 
-extern "C" BOOST_SYMBOL_EXPORT std::shared_ptr<abstract_plugin> get_plugin() {
+extern "C" BOOST_SYMBOL_EXPORT std::shared_ptr<abstract_plugin> plugin_factory() {
    return std::make_shared<track_and_trace_plugin>();
 }
 
+extern "C" BOOST_SYMBOL_EXPORT void options_setter(options_description& cli, options_description& cfg) {
+   track_and_trace_plugin::set_program_options(cli, cfg);
+}
+
```

Comment:
virtual method set_program_options(options_description& cli, options_description& cfg ) is not working when abstract_plugin is loaded as external lib or why is this "options_setter" exported ?